code/modules/projectiles/guns/ballistic/bow.dm

```diff
@@ -0,0 +1,102 @@
+/obj/item/gun/ballistic/bow
+	name = "bow"
+	desc = "A sturdy bow made out of wood and reinforced with iron."
+	icon_state = "bow_unloaded"
+	item_state = "bow"
+	var/icon_state_loaded = "bow_loaded"
+	var/icon_state_firing = "bow_firing"
+	var/item_state_loaded = "bow"
+	var/item_state_firing = "bow"
+	fire_sound = "bowshot" //
+	mag_type = /obj/item/ammo_box/magazine/internal/bow
+	//var/flags = HANDSLOW //
+	weapon_weight = WEAPON_HEAVY  //
+	var/draw_sound = 'sound/weapons/draw_bow.ogg'
+	var/ready_to_fire = 0
+	var/slowdown_when_ready = 2
+	casing_ejector = FALSE  //
+
+/obj/item/gun/ballistic/bow/update_icon()  ///SPECIAL THANKS TO OCULUS TO HELP ME WITH THIS.
+	if(ready_to_fire)
+		icon_state = icon_state_firing
+		item_state = item_state_firing
+		slowdown = slowdown_when_ready
+	else if(magazine.ammo_count() && !ready_to_fire)
+		icon_state = icon_state_loaded
+		item_state = item_state_loaded
+		slowdown = initial(slowdown)
+	else
+		icon_state = initial(icon_state)
+		item_state = initial(item_state)
+		slowdown = initial(slowdown)
+
```

Comment:
Could probably make a switch of some kind and instead of force-setting the slowdown, set it to be slowdown = slowdown + slowdown_when_ready.
for example
firing_status = 0, 1, 2, etc and then define the various states and work from there.

Careful when altering procs that are very high up the parent.